src/cast_core/src/CrudActor.cpp

```diff
@@ -239,16 +272,58 @@ auto createGenerator(YAML::Node source,
 
 }  // namespace
 
+struct CreateIndexOperation : public BaseOperation {
+    mongocxx::collection _collection;
+    metrics::Operation _operation;
+    DocumentGenerator _keysGenerator;
+    DocumentGenerator _indexOptionsGenerator;
+    mongocxx::options::index_view _operationOptions;
+    // TODO: convert from yaml
+
+    bool _onSession;
+
+    CreateIndexOperation(YAML::Node opNode,
+                         bool onSession,
+                         mongocxx::collection collection,
+                         metrics::Operation operation,
+                         PhaseContext& context,
+                         ActorId id)
+        : BaseOperation(context, opNode),
+          _collection(std::move(collection)),
+          _operation{operation},
+          _onSession{onSession},
+          _keysGenerator{createGenerator(opNode, "createIndex", "Keys", context, id)},
```

Comment:
This might just be me being petulant.... `createGenerator` makes me thinking we're generating the "create" command because `keysGenerator` generates... "keys" XD. Would `createDocumentGenerator` be a better name?

Reply:
lol ya good call  - renamed